onnx/defs/printer.cc

```diff
@@ -186,14 +186,17 @@ void ProtoPrinter::print(const TypeProto& type) {
     print(type.sparse_tensor_type());
 }
 
-void ProtoPrinter::print(const TensorProto& tensor) {
+void ProtoPrinter::print(const TensorProto& tensor, bool is_initializer) {
   output_ << PrimitiveTypeNameMap::ToString(tensor.data_type());
   if (tensor.dims_size() > 0)
     printSet("[", ",", "]", tensor.dims());
 
   if (!tensor.name().empty()) {
     output_ << " " << tensor.name();
   }
+  if (is_initializer) {
+    output_ << " = ";
+  }
```

Comment:
So if the given TensorProto is an initializer, the print function will just add a equal? Do you have an example what will be the printer like for initializer and non-initializer here?

Reply:
Yes. We are reusing the same code for two contexts. The syntax for initializer requires the " = ", but a TensorProto, used as an attribute value does not. The printer follows the exact syntax supported by the parser.

Example of initializer is here in [this parser test](https://github.com/onnx/onnx/blob/e1d921373fe58a22d62fa861daff1c22e279fb8a/onnx/test/cpp/parser_test.cc#L305).

Example of a non-initializer is here in [this parser test](https://github.com/onnx/onnx/blob/e1d921373fe58a22d62fa861daff1c22e279fb8a/onnx/test/cpp/parser_test.cc#L174)

And the parser test invokes the printer, and parses it back, to ensure round-tripping works [here](https://github.com/onnx/onnx/blob/e1d921373fe58a22d62fa861daff1c22e279fb8a/onnx/test/cpp/parser_test.cc#L23)